re-comitted project

28071233 opened the pull request:
-fixed github branch

28071233 committed:
re-comitted project

-fixed github branch

mindlyanarch commented on Game.cs:
```diff
@@ -10,16 +10,65 @@ internal class Game
+                while (choosing)
+                {
+                    Console.Write("Enter input here: ");
+                    string choice = Console.ReadLine();
```
Use Console.Readline().ToLower() so that it doesnt matter how the player capitalises it, it will still register the command

mindlyanarch commented on Game.cs:
```diff
@@ -10,16 +10,65 @@ internal class Game
+                        case "proceed":
+                            choosing = false;
+                            break;
+                        case "exit":
```
you should consider changing this command to "leave", it could be confused with the "quit" command

mindlyanarch commented on Game.cs:
```diff
@@ -10,16 +10,65 @@ internal class Game
+                    }
+                }
+
+                // Generate a number between 0 and 99 to use for creating the next room
```
This code needs to be in it's own function, currently it doesn't matter which command the player picks, the code will always generate a new room after each input

mindlyanarch commented on Player.cs:
```diff
@@ -13,9 +13,10 @@ public Player(string name, int health)
+        public static void PickUpItem(string item)
         {
-
+            //please work i need to sleep
```
mood

mindlyanarch commented on Program.cs:
```diff
@@ -12,8 +12,6 @@ static void Main(string[] args)
         {
             Game game = new Game();
             game.Start();
-            Console.WriteLine("Waiting for your Implementation");
```
You should probably give the player a message when they quit the program instead of just closing the window

mindlyanarch commented on Room.cs:
```diff
@@ -1,17 +1,95 @@
+            {
+                // Uses a randomly generated number to select what loot is present in the current room
+                Random randomNum = new Random();
+                int lootseed = randomNum.Next(0, 100);
```
Instead of using an if statement, Generate a random number with the length of the loot table as the maximum, and use that to index the loot table:

randNum = Rand.Next(LootTable.Length)

Player.PickupItem(LootTable[RandNum])

mindlyanarch commented on Room.cs:
```diff
@@ -1,17 +1,95 @@
+            {
+            // Contains contents of rooms in the format - Title : Description : has loot?
+                {"Empty Room", "A cold, damp room void of colour", "false"},
+                {"Treasure Room", "A cramped room composed of rotting wood and broken glass", "true" },
```
be careful to be within the character limit of 65 when creating these. I don't think any of these are longer, just something to keep in mind

mindlyanarch reviewed (approved):
Looks like you've managed to hit all of the features needed for a pass. You can improve your code by adding comments.

28071233 committed:
User's choice input sets all inputted characters to lower and removes white space

Added GenerateRoom() function

Renamed the "exit" command to "leave"

"quit" and "leave" are now functional and allows the program to terminate

Added message for when the program terminates

29576302 commented on Room.cs:
```diff
@@ -1,17 +1,95 @@
+            // Contains contents of rooms in the format - Title : Description : has loot?
+                {"Empty Room", "A cold, damp room void of colour", "false"},
+                {"Treasure Room", "A cramped room composed of rotting wood and broken glass", "true" },
+                {"Exit Room", "A lukewarm room with the sun seeping through the cracks", "false" }
```
Love the inclusion of room descriptions. Helps you visualise the environment despite the game being text-based. Maybe try looking for a different datatype for RoomIndex that allows you to store hasLoot as a boolean.

29576302 commented on Player.cs:
```diff
@@ -13,9 +13,10 @@ public Player(string name, int health)
+            //please work i need to sleep
+            //inventory.add(item);
         }
         public string InventoryContents()
```
Would be nice to see more done with the inventory and items. This could be a good opportunity to add more classes and explore inheritance.

29576302 commented on Room.cs:
```diff
@@ -1,17 +1,95 @@
+        private bool hasLoot;
 
-        public Room(string description)
+        public Room(int roomSeed)
+        {
+            // Uses a randomly generated number to select which room to load
+            if (roomSeed >= 90)
+            {
+                // copies contents from RoomIndex table and loads it into the Current room
+                this.title = RoomIndex[2, 0];
+                this.description = RoomIndex[2, 1];
+                this.hasLoot = bool.Parse(RoomIndex[2, 2]);
+            }
+            else if (roomSeed >= 60)
+            {
+                this.title = RoomIndex[1, 0];
+                this.description = RoomIndex[1, 1];
+                this.hasLoot = bool.Parse(RoomIndex[1, 2]);
+            }
+            else
+            {
+                this.title = RoomIndex[0, 0];
+                this.description = RoomIndex[0, 1];
+                this.hasLoot = bool.Parse(RoomIndex[0, 2]);
+            }
+        }
```
Could this be simplified by having the room seed range from 0-2 and indexing roomSeed rather than using an if/elif/else structure?

29576302 commented on Game.cs:
```diff
@@ -10,17 +10,73 @@ internal class Game
+        {
+            // Generate a number between 0 and 99 to use for creating the next room
+            Random randomNum = new Random();
+            int seed = randomNum.Next(0, 100);
```
Could this be simplified by having the room seed range from 0-2 and indexing roomSeed rather than using an if/elif/else structure?

29576302 commented on Game.cs:
```diff
@@ -10,17 +10,73 @@ internal class Game
+                while (choosing)
+                {
+                    Console.Write("Enter input here: ");
+                    string choice = Console.ReadLine().ToLower().Trim();
```
Good way to 'clean' the input.

29576302 reviewed (commented):
Looks pretty good overall. Would be nice to see the item and inventory aspects expanded upon.

28071233 committed:
Inventory works but i dont believe throwing "static void" into all my modified functions is good programming practice

28071233 committed:
Add comments throughout all files, lootseed has been renamed to lootSeed and added a divider for when new rooms are generated

28071233 committed:
Renamed "RoomIndex" to "roomIndex"

28071233 committed:
Added validation check for user's name
